package/batocera/core/batocera-configgen/configgen/configgen/utils/gamescope.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+import json
+import logging
+import subprocess
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from ..generators.Generator import Generator
+
+_logger = logging.getLogger(__name__)
+
+@contextmanager
+def set_gamescope_cmd(system: Emulator, CurrentResolution: Resolution, command: Command) -> Iterator[None]:
+
+    if not system.config.get_bool("gamescope"):
+        return
+
+    gamescope_cmd = ["/usr/bin/gamescope"]
+
+    # Retrieve output resolution from system options.
+    # if not defined, inherit from the current screen resolution
+    output_resolution = system.config.get_str("gamescope_output_resolution")
+    if output_resolution:
+        output_width, _, output_height = output_resolution.partition("x")
+    else:
+        output_width = str(CurrentResolution["width"])
+        output_height = str(CurrentResolution["height"])
+
+    gamescope_cmd.extend([
+        "-W", output_width.strip(),
+        "-H", output_height.strip()
+    ])
+
+    # Retrieve nested resolution from system options.
+    # default undefined inherit from gamescopt_output_resoluton
+    # the nested_resolution is upscaled|downscaled to output_resolution
+
+    nested_resolution = system.config.get_str("gamescope_nested_resolution")
+    if nested_resolution:
+        nested_width, _, nested_height = nested_resolution.partition("x")
+        gamescope_cmd.extend([
+            "-w", nested_width.strip(),
+            "-h", nested_height.strip()
+        ])
+
+    # Retrieve nested refresh rate.
+    # Default 60 if undefined
```

Comment:
Should we provide the default? If gamescope decides to change their default, our default will change unless we use one here.